pages/api/address.ts

```diff
@@ -0,0 +1,19 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { getEns } from './_utils'
+
+export default async function getEnsAddress(
+  request: NextApiRequest,
+  response: NextApiResponse
+) {
+  try {
+    const ensName = request.query.name
+    const ens = await getEns()
+    const address = await ens.name(ensName).getAddress()
+    if (!address) throw `No address found for "${ensName}"`
+
+    response.setHeader('Cache-Control', 's-maxage=86400')
```

Comment:
Vercel recommends to also add `max-age=0` as this deals with cache invalidation for redeployments as they write https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching